demo.md

```diff
@@ -0,0 +1,24 @@
+## Demo script:
+
+- ERC-4337 compatible Smart Contract wallet
+- Unique feature: gasless off-chain txs with instant finality enabled by HTLC technology, as used in Bitcoin Lightning
+- Setup required:
+  - wallet owners must designate an Intermediary in their SC wallet
+  - wallet owners and Intermediary must sign-off on some amount of wallet funds designated for off-chain txs
+  - run in-browser client to interact with SC wallet and peers
+  - `Explain browser tab setup with Alice(left), Irene(middle), Bob(right)`
+- How it works:
+  - on-chain txs are 2/2 multi-sig (user + intermediary)
+  - `Show on-chain tx from Alice to Metamask account`
+  - user can remove intermediary at any point to claim full ownership of wallet/funds
+  - `Point to 'Eject' button on UI`
+  - off-chain payments are multisig between payor, intermediary, and payee
+  - off-chain payments require users to be connected to the same intermediary (or have a path to each other through multiple intermediaries)
+  - `Show off-chain payment Alice to Bob`
+- Cross-chain payments
+  - works for off-chain payments only
+  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
+    - Alice designates some of her Network1 funds to Intermediary
+    - Bob designates some of his Network2 funds Intermediary
```

Comment:
"Alice has Network1 funds she can sign over to Irene, and Irene has Network2 funds she can sign over to Bob."

Reply:
I was trying to list requirements for off-chains payments to work. Can you rephrase your statement so that it indicates what MUST be done instead of what COULD be done?

Reply:
Does this satisfy what your are saying?
```
  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
    - Alice has funds on Network1
    - Intermediary has funds on Network2
    - Intermediary has an exchange rate to convert between Network1 funds and Network2 funds
    - Intermediary can access both Network1 and Network2 to monitor Alice & Bob balances
```

Reply:
Fair enough.

> Does this satisfy what you are saying?
> 
> ```
>   - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
>     - Alice has sufficient funds on Network1
>     - Intermediary has sufficient funds on Network2
I would specify that these funds need to be in Bob's wallet. Not just "on Network2"


Otherwise, seems good. I don't know if you need to underline that the intermediary can access both networks. That is probably implied by having funds on each network.

Reply:
Good call, I'll remove the bullet about intermediary accessing both network. How about this:
```
  - Alice on Network1 can pay Bob on Network2 if the following conditions are met:
    - Alice has sufficient funds on Network1
    - Bob has designated enough of his Network2 funds for Intermediary
    - Intermediary has an exchange rate to convert between Network1 funds and Network2 funds
```